mm2src/coins/utxo/utxo_common.rs

```diff
@@ -752,6 +752,34 @@ pub fn tx_size_in_v_bytes(from_addr_format: &UtxoAddressFormat, tx: &UtxoTx) ->
     }
 }
 
+/// Implements building utxo script pubkey for an address by the address format
+pub fn get_script_for_address(coin: &UtxoCoinFields, addr: &Address) -> Script {
+    match addr.addr_format {
+        UtxoAddressFormat::Standard => {
+            if addr.prefix == coin.conf.pub_addr_prefix && addr.t_addr_prefix == coin.conf.pub_t_addr_prefix {
+                return Builder::build_p2pkh(&addr.hash);
+            } else if addr.prefix == coin.conf.p2sh_addr_prefix && addr.t_addr_prefix == coin.conf.p2sh_t_addr_prefix {
+                return Builder::build_p2sh(&addr.hash);
+            }
+        },
+        UtxoAddressFormat::Segwit => {
+            return Builder::build_p2witness(&addr.hash);
+        },
+        UtxoAddressFormat::CashAddress {
+            network: _,
+            pub_addr_prefix,
+            p2sh_addr_prefix,
+        } => {
+            if pub_addr_prefix == coin.conf.pub_addr_prefix {
+                return Builder::build_p2pkh(&addr.hash);
+            } else if p2sh_addr_prefix == coin.conf.p2sh_addr_prefix {
+                return Builder::build_p2sh(&addr.hash);
+            }
```

Comment:
Same here

Reply:
fixed, added error
(btw this fn will be eliminated if my PR #1960 for output_script refactoring is merged)